### DISCUSSION TITLE
Enhancement: Global Prompts and User Management Function

### DISCUSSION THREAD

**Discussion by aoaim** [danny-avila/LibreChat#3082] [Feature Requests & Suggestions]
### What features would you like to see added?

Global Prompts and User Management Function

### More details

LibreChat is excellent, both in terms of performance and functionality. However, to be honest, it's hard not to compare it with another excellent frontend Open-WebUI while using it. I have both instances at the same time and encountered two issues during usage. 

The first issue is that LibreChat seems to lack the concept of a global prompt, which automatically inserts a system prompt before each conversation. While LibreChat has presets, they are not global and can only be used within a certain conversation segment. The benefit of this is automatically adding a premise to all AI conversations. 

The second issue is that LibreChat does not have user management functionality. Of course, this isn't a major problem either; I have been using my instance separately without the need for sharing externally.

Thanks.

### Which components are impacted by your request?

General

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Thanks for trying out LibreChat!

> it's hard not to compare it with another excellent frontend Open-WebUI while using it

Of course, they have borrowed and continue to borrow a lot of ideas we pioneered here as this project has been around longer. There is a lot of overlap in the user-base, too, and naturally there is some competition in the space. A major difference is that LibreChat is better designed to scale. We have several admins hosting instances for over 2,000 people, and scaling has been a bit of an afterthought for open-webui, as the primary "target" has been people with single or few user setups.

> While LibreChat has presets, they are not global and can only be used within a certain conversation segment. The benefit of this is automatically adding a premise to all AI conversations.

We have Model Specs, which function as "global prompts": 
https://www.librechat.ai/docs/configuration/librechat_yaml/object_structure/model_specs

For example using model specs, I can create an experience like this:

![image](https://github.com/danny-avila/LibreChat/assets/110412045/e3e3a447-bc93-4ee7-9011-bec0a774cfaf)
![image](https://github.com/danny-avila/LibreChat/assets/110412045/1eb3946e-5e5a-4573-945d-8d9136dce749)

> The second issue is that LibreChat does not have user management functionality. 

This is in the works, along with more admin-features from the frontend. This is one big feature that will be designed as such: https://github.com/danny-avila/LibreChat/issues/3023

**Reply by alexanderswerdlow** [reply]
Just adding a +1 to the global prompts feature request. The model specs are a decent alternative, but it would be nice to have an option that is more like the ChatGPT interface where it is on by default for all models with an option to disable it per model/chat. Model specs also seems to limit the entire interface which is a pretty big drawback to have to choose between the entire normal UI vs. a default prompt.

Anyway thanks for making such a great project! Just started using it and love it so far!

**Reply by danny-avila** [reply]
@alexanderswerdlow awesome thank you. 

I'm happy to say **global prompts are already implemented**, in the sense that "prompts" are understood in this app, which are pre-filled values for chat input you can click/access by keyboard. They are not pre-configured system instructions per endpoint/model.

I'm excited to say I'm working on an improvement on model specs based on all the feedback for it along with feedback from other features, please stay tuned!

**Reply by alexanderswerdlow** [reply]
After my comment, I found that out and it works great; I guess I just assumed it would clear automatically but didn't check😅

**Reply by ahmed2m** [reply]
I see that #3023 was done which is great and I love the new admin functions and would love to see it expand! but that ticket doesn't address OP's User Management Functionality suggestion, as in to list the users, create new ones, perhaps see/limit their usage, delete them etc.. from within the dashboard.

**Comment by R-Prady** [comment]
When it comes to scalability LibreChat easily wins. It’s very well thought and designed.
